src/components/LiquidityMining/Create/Steps/PreviewAndCreate/index.tsx

```diff
@@ -42,46 +97,130 @@ export default function PreviewAndCreate({
+    liquidityPair ? liquidityPair : undefined
+  )
+  const [simulatedValuePercentage, setSimulatedValuePercentage] = useState(0)
 
-  const getConfirmButtonMessage = () => {
-    if (!account) {
-      return 'Connect your wallet'
-    }
-    if (userBalance && reward && reward.greaterThan('0') && userBalance.lessThan(reward)) {
-      return 'Insuffucient balance'
+  const maxStakedSimulatedAmount = useMemo(() => {
+    const base = stakingCap
+      ? parseFloat(stakingCap.multiply(nativeTokenPrice.multiply(nativeCurrencyUSDPrice)).toSignificant(22))
+      : DOLLAR_AMOUNT_MAX_SIMULATION
+
+    const baseInUsd = parseFloat(nativeTokenPrice.multiply(nativeCurrencyUSDPrice).toFixed(22))
+
+    const baseValue = showUSDValue ? base : base / baseInUsd
+
+    const tokenOrPair = liquidityPair instanceof Token ? liquidityPair : liquidityPair?.liquidityToken
+
+    if (tokenOrPair && base !== 0 && baseInUsd !== 0) {
+      setSimulatedStakedAmount(
+        parseUnits(
+          calculatePercentage(base / baseInUsd, simulatedValuePercentage).toString(),
+          tokenOrPair.decimals
+        ).toString()
+      )
     }
-    return 'Deposit & create'
-  }
 
+    return calculatePercentage(baseValue, simulatedValuePercentage)
+  }, [
+    setSimulatedStakedAmount,
+    liquidityPair,
+    simulatedValuePercentage,
+    stakingCap,
+    nativeTokenPrice,
+    nativeCurrencyUSDPrice,
+    showUSDValue,
+  ])
+
+  const liquidityPercentChangeCallback = useCallback((value: number) => {
+    setSimulatedValuePercentage(value)
+  }, [])
+
+  const [innerLiquidityPercentage, setInnerLiquidityPercentage] = useDebouncedChangeHandler(
+    simulatedValuePercentage,
+    liquidityPercentChangeCallback,
+    10
+  )
   useEffect(() => {
     setAreButtonsDisabled(
-      !!(!account || !reward || !reward.token || reward.equalTo('0') || (userBalance && userBalance.lessThan(reward)))
+      !!(!account || !reward || !liquidityPair || !startTime || !endTime || approvals || campaign === null)
     )
-  }, [account, reward, userBalance])
+  }, [account, reward, liquidityPair, startTime, endTime, approvals, campaign])
+  const getConfirmButtonMessage = () => {
+    if (!account) return 'Connect your wallet'
+    else if (approvals) return 'Rewards not approved/Insufficient balance'
+    else if (campaign === null) return 'One of tokens not priced'
+
+    return 'Deposit & create'
+  }
+  const isSingleSided = campaign instanceof SingleSidedLiquidityMiningCampaign
 
+  const handleUSDValueClick = useCallback(() => {
+    setShowUSDValue(!showUSDValue)
+  }, [showUSDValue])
   return (
     <Flex flexDirection="column" style={{ zIndex: -1 }}>
+      {campaign !== null && !loadingNativeCurrencyUsdPrice && (
+        <CampaignDetailWrapper flexDirection={'row'} mb="32px">
+          <StyledCampaignCard
+            token0={
+              campaign instanceof SingleSidedLiquidityMiningCampaign
```

Comment:
Use the defined variable `isSingleSided` here please

Reply:
Yeah for some reason it doesn't take into account evaluation I do before if campaign is undefined so it stops working...I tried debugging it but couldn't get it to work